core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatEntry.scala

```diff
@@ -430,61 +438,85 @@ class ColumnFormatValue extends SerializedDiskBuffer
   }
 
   private def decompressValue(buffer: ByteBuffer, onlyIfStored: Boolean): ColumnFormatValue = {
-    if (this.decompressionState != 0) {
+    var position = 0
+    var typeId = 0
+    var context: RegionEntryContext = null
+    var isDirect = false
+    var doReplace = false
+    var updateStats = false
+    val doDecompress = synchronized(if (this.decompressionState != 0) {
```

Comment:
The bracket after synchronized looks odd. Looks like its synchronizing on the return value of the expression. Is that the intention?

Reply:
No, synchronized is always on the object ("this" in this case) e.g. obj.synchronized.

A bit odd usage to minimize indentation changes.